### ISSUE TITLE
Remove Eslint Disabling Comments from Aio Examples

### ISSUE THREAD

**Issue by dario-piotrowicz** [angular/angular#43788]
As discussed [here](https://github.com/angular/angular/pull/43218#discussion_r693227877) and [here](https://github.com/angular/angular/pull/43218#pullrequestreview-744253351) the eslint disabling comments present in the aio examples should be ignored/removed so that they can still present in the source code but not displayed in the angular.io docs.

___

I have started looking into this and seems that we could change the logic in the [region-parser file](https://github.com/angular/angular/blob/1986407ab5939a1450be0ddecaa436e9dfeec052/aio/tools/transforms/examples-package/services/region-parser.js) so that we remove the lines in the code regions
(it does work, not sure if this is the right place where to put this logic)

I wanted to open this issue to discuss a bit the requirement as this can range from being an easy task to being much more complicated and I was wondering what we would want to achieve

___

First of all I just wanted to mention that in the file I mentioned, the lines are handled one by one:
https://github.com/angular/angular/blob/1986407ab5939a1450be0ddecaa436e9dfeec052/aio/tools/transforms/examples-package/services/region-parser.js#L42
 and my idea was to follow insert the login here

I started playing around with removing single line comments such as:
```ts
// eslint .*
```
and 
```html
<!-- eslint .* -->
```
and this can be accomplished easily enough with well defined regexes

But then I thought that multiline comments can be present too such as:
```ts
/* eslint-disable @angular-eslint/no-input-rename,
                  @angular-eslint/no-inputs-metadata-property,
                  @angular-eslint/no-output-rename,
                  @angular-eslint/no-outputs-metadata-property */
```
and this could be handled with starting and ending regexes/checks with also a boolean flag so that we can ignore the contained lines

so far it doesn't look that bad

The issue is that then I came to think of lines which mix both eslint disabling rules and actual code as:
```ts
const res = eval('42'); // eslint-disable-line no-eval
```
or much worse:
```ts
/* eslint-disable @angular-eslint/no-input-rename,
                  @angular-eslint/no-inputs-metadata-property,
                  @angular-eslint/no-output-rename,
                  @angular-eslint/no-outputs-metadata-property */ import {...
```

these last ones combined with the previous ones could end up involving some complex logic... :sweat: 

____

So basically I just wanted to open a discussion on what we would like to do here,
 - should this be 100% flexible and handle any possible edge case? (could require quite some time and logic behind it)
  - could this handle only single line comments for simplicity? (and throw errors if multiLine ones are used)
  - could this handle single + multiLine comments (mixed complexity, not too bad) but not lines with a mix of  both comment and code (and throw errors if mixed are used)
  - is the region parser the right place where to do this? (with the eslint specific logic held in a separate file)
  - is this worth doing vs keeping the eslint disabling comment out of the docregions/examples?

___

@gkalpak any input will be greatly appreciated! :smiley:
